src/tests/pools/D3MAaveV3NSTNoSupplyCapTypePool.t.sol

```diff
@@ -0,0 +1,332 @@
+        dai = dai_;
+    }
+
+    function join(address usr, uint256 amt) external {
+        dai.transferFrom(usr, address(this), amt);
+    }
+
+    function exit(address usr, uint256 amt) external {
+        dai.transfer(usr, amt);
+    }
+    
+}
+
+contract NstJoinMock {
+
+    TokenMock public nst;
+
+    constructor(TokenMock nst_) {
+        nst = nst_;
+    }
+
+    function join(address usr, uint256 amt) external {
+        nst.transferFrom(usr, address(this), amt);
+    }
+
+    function exit(address usr, uint256 amt) external {
+        nst.transfer(usr, amt);
+    }
+    
+}
+
+contract D3MAaveV3NSTNoSupplyCapTypePoolTest is D3MPoolBaseTest {
+
+    AToken anst;
+    FakeLendingPool aavePool;
+    DaiJoinMock daiJoin;
+    TokenMock nst;
+    NstJoinMock nstJoin;
+    
+    D3MAaveV3NSTNoSupplyCapTypePool pool;
+
+    function setUp() public {
+        baseInit("D3MAaveV3NoSupplyCapTypePool");
+
+        nst = new TokenMock(18);
+        anst = new AToken(18);
+        daiJoin = new DaiJoinMock(dai);
+        nstJoin = new NstJoinMock(nst);
+        anst.mint(address(this), 1_000_000 ether);
+        aavePool = new FakeLendingPool(address(anst), address(nst));
+        anst.rely(address(aavePool));
+
+        dai.mint(address(daiJoin), 1_000_000 ether);
+        nst.mint(address(nstJoin), 1_000_000 ether);
+
+        setPoolContract(pool = new D3MAaveV3NSTNoSupplyCapTypePool("", address(hub), address(nstJoin), address(daiJoin), address(aavePool)));
+    }
+
+    function test_constructor_sets_values() public {
+        assertEq(address(pool.nstJoin()), address(nstJoin));
+        assertEq(address(pool.nst()), address(nst));
+        assertEq(address(pool.daiJoin()), address(daiJoin));
+        assertEq(address(pool.dai()), address(dai));
+    }
+
+    function test_can_file_king() public {
+        assertEq(pool.king(), address(0));
+
+        pool.file("king", address(123));
+
+        assertEq(pool.king(), address(123));
+    }
+
+    function test_cannot_file_king_no_auth() public {
+        pool.deny(address(this));
+        assertRevert(address(pool), abi.encodeWithSignature("file(bytes32,address)", bytes32("king"), address(123)), "D3MAaveV3NoSupplyCapTypePool/not-authorized");
+    }
+
+    function test_cannot_file_king_vat_caged() public {
+        vat.cage();
+        assertRevert(address(pool), abi.encodeWithSignature("file(bytes32,address)", bytes32("king"), address(123)), "D3MAaveV3NoSupplyCapTypePool/no-file-during-shutdown");
+    }
+
+    function test_deposit_calls_lending_pool_deposit() public {
+        TokenMock(address(anst)).rely(address(aavePool));
+        dai.mint(address(pool), 1);
+        vm.prank(address(hub)); pool.deposit(1);
+        (address asset, uint256 amt, address dst, uint256 code) = FakeLendingPool(address(aavePool)).lastDeposit();
+        assertEq(asset, address(nst));
+        assertEq(amt, 1);
+        assertEq(dst, address(pool));
+        assertEq(code, 0);
+    }
+
```

Comment:
Any reason to not have copied the withdraw tests?

```
    function test_withdraw_calls_lending_pool_withdraw() public {
        // make sure we have Nst to withdraw
        TokenMock(address(nst)).mint(address(aavePool), 1);

        vm.prank(address(hub)); pool.withdraw(1);
        (address asset, uint256 amt, address dst) = FakeLendingPool(address(aavePool)).lastWithdraw();
        assertEq(asset, address(nst));
        assertEq(amt, 1);
        assertEq(dst, address(pool));
    }

    function test_withdraw_calls_lending_pool_withdraw_vat_caged() public {
        // make sure we have Nst to withdraw
        TokenMock(address(nst)).mint(address(aavePool), 1);

        vat.cage();
        vm.prank(address(hub)); pool.withdraw(1);
        (address asset, uint256 amt, address dst) = FakeLendingPool(address(aavePool)).lastWithdraw();
        assertEq(asset, address(nst));
        assertEq(amt, 1);
        assertEq(dst, address(pool));
    }
```

Reply:
They were failing, and I don't find these unit tests provide much value since the mocks are pretty divergent from the way the actual code works. The integration tests are providing most of the confidence here.

Reply:
Ok yeah, anyway those are working ^, I adapted a bit.

Reply:
ah ty, re-added